### ISSUE TITLE
Toggling editor scale to Aspect takes a long time

### ISSUE THREAD

**Issue by Epihaius** [fireclawthefox/DirectGuiDesigner#1]
Hi,

When I press the toolbutton to switch editor scale to Pixel, the change is pretty much instantaneous. But when I switch back to Aspect, this takes up an entire second on my laptop and about half a second on my desktop PC.
That's with a completely empty layout, so I find this lag a bit surprising.
To clarify, the entire application seems to freeze during that time; the button tooltip doesn't disappear and the interface is unresponsive.

Do you have any idea what might be causing this?

Tested on Windows 10

**Comment by fireclawthefox** [comment]
Hi,

sorry for taking so long to respond, seems I haven't been notified about your issue reports... Anyways, yeah, I am aware of this behavior, but I don't know where it comes from yet. It wasn't a high priority for me as I expected people would switch only once at the beginning of editing a GUI. I'm not even sure if it's fully possible to use both scales at the same time with the editor. But will take a look at it and see if I can enhance the editor performance in general.

**Comment by Epihaius** [comment]
After some investigating, the culprit turns out to be the `DirectGrid` object.
In [this line of code](https://github.com/fireclawthefox/DirectGuiDesigner/blob/1ba0a3eb4b14f29f972ca1c05428a79e1dcb5e11/DirectGuiDesignerEditorCanvas.py#L178), when I change the grid spacing to a much smaller value like e.g. `0.005`, the lag accordingly increases tenfold, more or less.
It seems that DirectGrid [regenerates its geometry](https://github.com/panda3d/panda3d/blob/371e60c768cfd1a1383c5b05f1cab2078014ccea/direct/src/directtools/DirectGrid.py#L75) when `setGridSpacing` is called on it, creating as many grid lines as needed, depending on the given value, as well as **on the current grid size**.

So the solution seems quite simple: change the order of the calls to `setGridSpacing` and `setGridSize`; this made the lag disappear for me.
(Note that the order of these calls should not be changed in the code that switches from `aspect2d` to `pixel2d`.)

Hopefully that fixes this issue :) .

**Comment by fireclawthefox** [comment]
Ah, yes, I also found that part. I've changed it a little differently though. In the new version I'm seting the grid size to 1 so it doesn't need to calculate as much at all in those calls and then set the size back to the desired value, which enhances the performance even a little more. Sadly i haven't had time to commit those changes yesterday, but I'll try to get the fix for this and your other issue up today.
Anyways, thanks for taking a look into this too.
